src/createApp.ts

```diff
@@ -899,26 +914,21 @@ const createApp = compose({
 			// custom elements.
 			const factories = widgetFactories.get(this);
 			const instances = widgetInstances.get(this);
-			return new Promise((resolve) => {
-				// First see if a factory exists for the widget.
-				let factory: WidgetFactory;
-				try {
-					factory = factories.get(id);
-				} catch (missingFactory) {
-					try {
-						// Otherwise try and get an existing instance.
-						const instance = instances.get(id);
-						resolve(instance);
-						return; // Make sure to return!
-					} catch (_) {
-						// Don't confuse people by complaining about missing instances, rethrow the
-						// original error.
-						throw missingFactory;
-					}
-				}
-				// This is only reached when a factory exists. Call it and resolve with the result.
-				// If it throws that's fine, it'll reject the promise.
-				resolve(factory());
+
+			let missingFactory: any;
+			return Promise.resolve().then(() => {
+				const factory = factories.get(id);
+				return factory();
+			})
+			.catch((e) => {
+				missingFactory = e;
```

Comment:
The error is actually meaningful from the first catch as it can be more than just unable to find `id` in registry... Perhaps we should only create a new `Error` if the original error is `Could not find a value for identity '${id}'` as then it would be adding value.